Assets/General/Scripts/Story/StoryManager.cs

```diff
@@ -0,0 +1,176 @@
+/*****************************************************************************
+// Name :           StoryManager.cs
+// Author :         Charlie Polonus
+// Created :        11/6/2024
+// Description :    Handles the story beats and story beat events for during
+                    the game
+*****************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class StoryManager : MonoBehaviour
+{
+    // The singleton StoryManager
+    public static StoryManager main;
+
+    // The StoryBeat and coroutine system
+    public List<StoryBeat> StoryBeats;
+    private Coroutine _beatEventsCoroutine;
+
+    // Editor settings
+    public int OpenStoryBeat;
+
+    /// <summary>
+    /// Run initialization functions on the StoryManager
+    /// </summary>
+    private void Awake()
+    {
+        if (main == null)
```

Comment:
Just to keep awake clean can we make the establishing of a singleton its own function?